test/python/circuit/test_parameters.py

```diff
@@ -1277,6 +1277,12 @@ class TestParameterExpressions(QiskitTestCase):
 
     supported_operations = [add, sub, mul, truediv]
 
+    def setUp(self):
+        super().setUp()
+        # TODO: delete once bind_parameters is removed from the codebase
+        #  and related tests are also removed.
+        warnings.filterwarnings("ignore", category=DeprecationWarning)
```

Comment:
This looks great thanks.  I think we can even be a bit stricter here - I think if we set the `module` field to `r"test\.python\.circuit\.test_parameters"`, we'd be filtering _only_ warnings that are emitted by direct calls to `bind_parameters` in this test suite, and it'd still fail if we triggered any calls to `bind_parameters` internally.